src/Components/sharedComponents/ImageBuilderHeader.tsx

```diff
@@ -15,16 +15,54 @@ import {
   PageHeader,
   PageHeaderTitle,
 } from '@redhat-cloud-services/frontend-components';
+import { addNotification } from '@redhat-cloud-services/frontend-components-notifications/redux';
+import { useDispatch } from 'react-redux';
 
+import { useComposeBlueprintMutation } from '../../store/imageBuilderApi';
+import { HttpErrorList } from '../../store/imageBuilderApi';
 import './ImageBuilderHeader.scss';
 
 type ImageBuilderHeaderPropTypes = {
   experimentalFlag?: string | true | undefined;
+  selectedBlueprint?: string | undefined;
 };
 
 export const ImageBuilderHeader = ({
   experimentalFlag,
+  selectedBlueprint,
 }: ImageBuilderHeaderPropTypes) => {
+  const [
+    buildBlueprint,
+    { error: imageBuildError, isLoading: imageBuildLoading },
+  ] = useComposeBlueprintMutation();
+  const dispatch = useDispatch();
+  useEffect(() => {
+    if (imageBuildError && 'status' in imageBuildError) {
+      const errors = imageBuildError.data as HttpErrorList;
+      const errMsg = errors.errors.map((e) => e.detail).join(', ');
+      dispatch(
+        addNotification({
+          variant: 'danger',
+          title: 'Error building blueprint',
+          description: errMsg,
+        })
+      );
+    }
+  }, [imageBuildError]);
+
+  const onBuildHandler = async () => {
+    const buildResponse =
+      selectedBlueprint && (await buildBlueprint({ id: selectedBlueprint }));
+    if (buildResponse && 'error' in buildResponse === false) {
+      dispatch(
```

Comment:
Check out how we handle dispatching notifications in the `enhancedImageBuilderApi.ts` file. 

You can add a field for `buildBlueprint` and follow the same pattern that we use for `composeImage` or `cloneCompose` to dispatch the notification.

Reply:
We will want to do it with an enhancement because we will use tag invalidation to trigger refetching the images associated with the blueprint. This lets us handle tag invalidation and dispatching either the success or error notification all in the same place.

Reply:
Thanks. as we discussed offline, using `enhanceEndpoints` for cache management (or  future optimistic updates as follow-up) makes sense 👍 , I'd use the component level for taking care of error management though, but let's keep code consistency